### PR TITLE
Room-Events: edit option

### PR DESCRIPTION
**Pull request by sparkychildcharlie**
Follow up to https://github.com/Zarel/Pokemon-Showdown/pull/3669

### CODE REVIEW

File: chat-plugins/room-events.js [hunk 1]
```diff
@@ -46,12 +47,18 @@ exports.commands = {
 			const eventId = toId(eventName);
 			if (!eventId) return this.errorReply("Event names must contain at least one alphanumerical character.");
 
+			if (room.events[eventId] && this.cmd === 'add') {
```
Comment (Zarel) [inline chat-plugins/room-events.js]: Do the other way around, too: `edit` shouldn't work if the room event doesn't already exist

Comment (sparkychildcharlie) [inline chat-plugins/room-events.js]: oh my bad ooops

File: chat-plugins/room-events.js [hunk 2]
```diff
@@ -46,12 +47,22 @@ exports.commands = {
...
+				this.errorReply('The event already exists.  If you want to overwrite this event, use /roomevents edit.');
+				this.sendReplyBox(Chat.html `<code>/roomevents edit ${room.events[eventId].eventName} | ${room.events[eventId].date} | ${room.events[eventId].desc}</code>`);
```
Comment (Zarel) [inline chat-plugins/room-events.js]: No space between `html` and `` ` ``

File: chat-plugins/room-events.js [hunk 3]
```diff
@@ -46,12 +47,22 @@ exports.commands = {
...
+				return;
+			} else if (this.cmd === 'edit' && !room.events[eventId]) {
+				this.errorReply('This event does not exist; use /roomevents add.');
```
Comment (Zarel) [inline chat-plugins/room-events.js]: "There's no event with the name `name`."

File: chat-plugins/room-events.js [hunk 4]
```diff
@@ -46,12 +47,22 @@ exports.commands = {
 			const eventId = toId(eventName);
 			if (!eventId) return this.errorReply("Event names must contain at least one alphanumerical character.");
 
+			if (room.events[eventId] && this.cmd === 'add') {
+				this.errorReply('The event already exists.  If you want to overwrite this event, use /roomevents edit.');
```
Comment (Zarel) [inline chat-plugins/room-events.js]: "There's already an event named `name`. If you want to replace that event,"

File: chat-plugins/room-events.js (hunk 2 above)
Comment (sparkychildcharlie) [inline chat-plugins/room-events.js]: oops i just copied bumba's styling

Comment (Zarel) [inline chat-plugins/room-events.js]: ...someone should tell @bumbadadabum not to do that, either...

**Pull request merged**